GoodVibes.Client.Lovense/Models/Abstractions/LovenseToy.cs

```diff
@@ -148,41 +176,29 @@ private void AddCommandToList(LovenseCommandEnum function, int value)
             {
                 values!.Add(value);
             }
+
+            CommandReceived = true;
         }
 
         public string? GetCommandString()
         {
-            if (ToyCommands.Count == 0) return null;
-            
-            var commandStr = string.Empty;
-            foreach (var toyCommand in ToyCommands)
-            {
-                //var highestValue = 0;
-                var lastValue = 0;
-                if (toyCommand.Key == Function1 || toyCommand.Key == Function2)
-                {
-                    var values = toyCommand.Value;
+            if (!CommandReceived) return null;
 
-                    // It seems like this, if you're quick is actually null.
-                    if (values == null) continue;
-
-                    // TODO: Add more calculation methods for different behaviors
-                    //highestValue = values.Prepend(0).Max();
-                    lastValue = values.Last();
-                }
-                else
-                {
-                    continue;
-                }
+            var functions = new[] { Function1, Function2, Function3 };
+            var commands = new List<string>();
+            foreach (var function in functions)
+            {
+                var commandFound = ToyCommands.TryGetValue(function, out var values);
+                if (!commandFound || values == null) continue;
 
-                if (commandStr != string.Empty)
+                if (function != LovenseCommandEnum.None)

```

Comment:
Could you put this earlier to skip over having to grab the value for this?

Reply:
Definitely could and I just did :)